src/Validation/Rules/After.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Validation\Rules;
+
+use Attribute;
+use DateTimeImmutable;
+use DateTimeInterface;
+use Tempest\Validation\Rule;
+use Throwable;
+
+#[Attribute]
+final readonly class After implements Rule
```

Comment:
I think `AfterDate` might be a clearer name, but also sounds a bit… clunky. Unsure about it

Reply:
Yeah I wasn't a fan of either After or AfterDate, but I don't really have a better idea. Future or FutureDate was also something that I've considered but it feels more confusing for some reason

Reply:
Let's rename it to `AfterDate` and `BeforeDate` for now, and I'll try to come up with a better name once this is merged :)